android/src/main/java/com/lykhonis/flutterbilling/BillingPlugin.java

```diff
@@ -164,17 +170,57 @@ public void failed() {
         });
     }
 
+    private void subscribe(final String identifier, final Result result) {
+        executeServiceRequest(new Request() {
+            @Override
+            public void execute() {
+              final int subscriptionSupportResponse = billingClient.isFeatureSupported(FeatureType.SUBSCRIPTIONS);
+              if (BillingResponse.OK != subscriptionSupportResponse) {
+                    result.error("NOT SUPPORTED", "Subscriptions are not supported.", null);
+                    return;
+              }
+
+              final int responseCode = billingClient.launchBillingFlow(
+                      activity,
+                      BillingFlowParams.newBuilder()
+                              .setSku(identifier)
+                              .setType(SkuType.SUBS)
+                              .build());
+
+              if (responseCode == BillingResponse.OK) {
+                  pendingPurchaseRequests.put(identifier, result);
+              } else {
+                  result.error("ERROR", "Failed to subscribe with error " + responseCode, null);
+              }
+            }
+
+            @Override
+            public void failed() {
+              result.error("UNAVAILABLE", "Billing service is unavailable!", null);
+            }
+        });
+    }
+
     private void fetchPurchases(final Result result) {
         executeServiceRequest(new Request() {
             @Override
             public void execute() {
-                final Purchase.PurchasesResult purchasesResult = billingClient.queryPurchases(SkuType.INAPP);
-                final int responseCode = purchasesResult.getResponseCode();
+                final Purchase.PurchasesResult productResult = billingClient.queryPurchases(SkuType.INAPP);
+                final int productResponseCode = productResult.getResponseCode();
 
-                if (responseCode == BillingResponse.OK) {
-                    result.success(getIdentifiers(purchasesResult.getPurchasesList()));
+                final Purchase.PurchasesResult subscriptionResult = billingClient.queryPurchases(SkuType.SUBS);
+                final int subscriptionResponseCode = subscriptionResult.getResponseCode();
+
+                if (productResponseCode == BillingResponse.OK) {
```

Comment:
Consider error case, there should be single `if` condition checking `BillingResponse.OK` for both products and subscriptions, otherwise error.